### PR TITLE
gameOfWar not quite working, but here it is

### PR DESCRIPTION
**Pull request by nclevine**

### CODE REVIEW

File: cards.js [hunk 1]
```diff
@@ -17,7 +17,116 @@ function shuffle(array){
...
+
+function gameOfWar(deck){
+  var turnsTaken = 0;
+  var p1Deck = [];
+  var p2Deck = [];
+  while(deck.length > 0){
+    p1Deck.push(deck.shift());
+    p2Deck.push(deck.shift());
+  };
+  while(p1Deck.length > 0 && p2Deck.length > 0){
```
Comment (jrhorn424) [inline cards.js:83]: As written, as soon as `p1Deck.length === 0`, the loop will stop regardless of `p2Deck.length`. If that's a problem, you could use a logical OR instead: `while(p1Deck.length > 0 || p2Deck.length > 0){`.

File: cards.js [hunk 2]
```diff
@@ -17,7 +17,116 @@ function shuffle(array){
...
+        var p1TieDiscards = []
+        p1TieDiscards.push(p1Deck.shift(), p1Deck.shift());
+        var p2TieDiscards = []
+        p2TieDiscards.push(p2Deck.shift(), p2Deck.shift());
+        var tieCard1 = p1Deck.shift();
+        var tieCard2 = p2Deck.shift();
+        if (faceCardValue(card1.value) > faceCardValue(card2.value)){
+          winner = tieCard1;
+          p1Deck.push(card1, card2, tieCard1, tieCard2);
+          Array.prototype.push.apply(p1Deck, p1TieDiscards);
```
Comment (jrhorn424) [inline cards.js:105]: Any reason you opted to use `Array.prototype.push.apply(p1Deck...` rather than `p1Deck.push(p1TieDiscards)`?

**Comment by jrhorn424** [comment]
Could you elaborate a bit on what wasn't working when you submitted?

**Comment by jrhorn424** [comment]
Your code is very readable and clean. I like the consistent indentation and your names are very descriptive. :+1:

If you were to work on this again, I'd suggest reducing some of the duplication present in the main Game of War function by extracting methods. As a bonus challenge, I'd work on un-nesting while loops as much as possible.

Good work!

**Comment by nclevine** [comment]
What wasn't working was that sometimes/somehow cards got left out of the
pile, and there ended up being less than 52 total in play (I think this
happened whenever there was a tie and people had to throw down three extra
cards).

The reason I was using Array.prototype.push.apply(..., ...) was to join two
arrays together so that everything stayed at the same index level. That is
to say, since the extra cards pulled out for a tie scenario were put into
their own mini array, if I had just push that array back into the player's
deck, it would act as an array within an array.

ex /
deck = [card, card, card];
extraCards = [card, card];
deck.push(extraCards); // deck == [card, card, card, [card, card]], but
what i want is deck == [card, card, card, card, card]

I'll have to look at it more in depth to see what's going on with the
missing cards. But another thing is that it doesn't seem like the game ever
has any hope of ending. Maybe that's just how the game of war is .. haha,
but neither "player" seemed to have any likelihood of gaining the majority
of the deck- they sort of alternated between having a few more or less than
the other and it fluctuated in some small range of difference...

**Pull request closed without merging**